server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -204,7 +222,11 @@ public void onAssignment(
                 topicIdPartition, directoryId, nowNs, successCallback);
         ready.put(topicIdPartition, assignment);
         if (log.isTraceEnabled()) {
-            log.trace("Registered assignment {}: {}", assignment, reason);
+            log.trace("Registered assignment {}: {}, moving {} into {}",
+                assignment,
+                reason,
+                topicIdToDescription.apply(assignment.topicIdPartition().topicId()),
```

Comment:
Since we now have `metadataImageSupplier`, we no longer need the `topicIdToDescription ` constructor parameter. 

```java
Optional.ofNullable(
        metadataImageSupplier.get().topics()
                .getTopic(assignment.topicIdPartition().topicId())
    )
    .map(TopicImage::name)
    .orElse("[unknown topicid path]");
```